### ISSUE TITLE
Is that possible to make a survival model from xgboost?

### ISSUE THREAD

**Issue by ajing** [dmlc/xgboost#386]
Basically, change the objective function and evaluation metrics. The evaluation metrics could be [timeROC](http://cran.r-project.org/web/packages/timeROC/index.html).

**Comment by BrianMiner** [comment]
You can def create a discrete time hazard model without any modifications, using the proper person-period data set. http://gseacademic.harvard.edu/alda/Handouts/ALDA%20Chapter%2011.pdf is a quick reference. SAS actually has a fantastic course with a lot of details (can be ran on my software).

**Comment by ajing** [comment]
I am not sure how to combine period-event as one dependent variable.

**Comment by BrianMiner** [comment]
Does a discrete time model fit your data or do you have continuous time? The work I do deals with monthly data mainly so it is a perfect fit.

**Comment by ajing** [comment]
I have death flag as the survival outcome variable, and last known alive day (in days) as time to event.

**Comment by BrianMiner** [comment]
If you can treat the problem as a discrete time hazard model (read the pdf I posted) you can use any supervised learning algorithm (including xgboost). I have fit them with GBM and actually am doing it as we speak. If discrete time is not appropriate then I suggest you use another method such as cox regression (which is actually a loss function in the R GBM package if you wanted to stick with boosted trees). Good luck!

**Comment by mpearmain** [comment]
+1 cox regression loss function

**Comment by tqchen** [comment]
move objective function proposal to https://github.com/dmlc/xgboost/issues/749
